bin/bundle_tools/staging.py

```diff
@@ -0,0 +1,210 @@
+import os
+import urllib3
+from urllib3.util import parse_url
+from shutil import copyfileobj
```

Comment:
Consider separating first-party imports from stdlib imports (and third-party imports, of which there are none here) by an extra newline.